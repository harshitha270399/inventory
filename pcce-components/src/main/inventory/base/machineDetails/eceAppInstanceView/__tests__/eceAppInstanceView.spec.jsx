import React from 'react';
import { shallow } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';
import ECEAppInstanceView from '../eceAppInstanceView'; 

const defaultProps = {
    updatedValues: { eceAppInstance: true },
    model: jest.fn()
};

describe('ECEAppInstanceView test suite', () => {
    it('should render correctly with eceAppInstance', () => {
        const component = shallow(<ECEAppInstanceView {...defaultProps} />);
        expect(EnzymeToJson(component)).toMatchSnapshot();
    });

    it('should not render when eceAppInstance is not provided', () => {
        const propsWithoutECE = {
            ...defaultProps,
            updatedValues: { eceAppInstance: false }
        };

        const component = shallow(<ECEAppInstanceView {...propsWithoutECE} />);
        expect(EnzymeToJson(component)).toMatchSnapshot();
    });
});
