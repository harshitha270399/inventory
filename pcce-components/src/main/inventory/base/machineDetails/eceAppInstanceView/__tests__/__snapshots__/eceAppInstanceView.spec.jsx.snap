// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ECEAppInstanceView test suite should not render when eceAppInstance is not provided 1`] = `
<div>
  <div />
  <div
    className="flex mar-top-125r flex0"
  >
    <div
      className="flex0-170basis"
    >
      <label
        className="control-label"
        htmlFor="appInstanceInput"
      >
        <span
          className="required"
        >
          *
        </span>
      </label>
    </div>
    <div />
  </div>
</div>
`;

exports[`ECEAppInstanceView test suite should render correctly with eceAppInstance 1`] = `
<div>
  <div />
  <div
    className="flex mar-top-125r flex0"
  >
    <div
      className="flex0-170basis"
    >
      <label
        className="control-label"
        htmlFor="appInstanceInput"
      >
        <span
          className="required"
        >
          *
        </span>
      </label>
    </div>
    <div>
      <Input
        className=""
        id="appInstanceInput"
        inputStyle="normal"
        maxLength={24}
        type="text"
      />
    </div>
  </div>
</div>
`;
